specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+
+## .NET, WinRT
+```c#
+    webView.CoreWebView2.FrameCreated += (sender, args) =>
+    {
+        args.Frame.DOMContentLoaded += (frameSender, DOMContentLoadedArgs) =>
+        {
+            args.Frame.ExecuteScriptAsync(
+                "let content = document.createElement(\"h2\");" +
+                "content.style.color = 'blue';" +
+                "content.textContent = \"The url of the iframe is \" + window.location.href.toString();" +
+                "document.body.appendChild(content);");
+        };
+    };
+```
+
+## Win32 C++
+```cpp
+    CHECK_FAILURE(m_appWindow->GetWebView()->QueryInterface(IID_PPV_ARGS(&m_webView2)));
+    wil::com_ptr<ICoreWebView2_4> webview2_4 = m_webView.try_query<ICoreWebView2_4>();
+    if (webview2_4)
+    {
+        CHECK_FAILURE(webview2_4->add_FrameCreated(
+            Callback<ICoreWebView2FrameCreatedEventHandler>(
+                [](ICoreWebView2* sender, ICoreWebView2FrameCreatedEventArgs* args)
+                    -> HRESULT {
+                    wil::com_ptr<ICoreWebView2Frame> webviewFrame;
+                    CHECK_FAILURE(args->get_Frame(&webviewFrame));
+                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental =
+                        webviewFrame.try_query<ICoreWebView2ExperimentalFrame>();
+                    if (frameExperimental)
+                    {
+                        frameExperimental->add_DOMContentLoaded(
+                            Callback<
+                                ICoreWebView2ExperimentalFrameDOMContentLoadedEventHandler>(
+                                [](
+                                    ICoreWebView2Frame* frame,
+                                    ICoreWebView2DOMContentLoadedEventArgs* args) -> HRESULT {
+                                    wil::com_ptr<ICoreWebView2ExperimentalFrame> frameExperimental;
+                                    frame->QueryInterface(IID_PPV_ARGS(&frameExperimental));
+
+                                    frameExperimental->ExecuteScript(
+                                        LR"~(
+                                        let content = document.createElement("h2");
+                                        content.style.color = 'blue';
+                                        content.textContent = "The url of the iframe is " + window.location.href.toString();
+                                        document.body.appendChild(content);
+                                        )~",
+                                        Callback<ICoreWebView2ExecuteScriptCompletedHandler>(
+                                            [](HRESULT error, PCWSTR result) -> HRESULT {
+                                                return S_OK;
+                                            })
+                                            .Get());
+                                    return S_OK;
+                                })
+                                .Get(),
+                            NULL);
```

Comment:
Is the registration token really optional? In the IDL, the registration token is marked `[out]`, which disallows NULL.

Reply:
@david-risney ensure win32/com doc notes this can be nullptr despite IDL not having a way to express that.